mriqc/cli/run.py

```diff
@@ -30,14 +29,30 @@ def main():
     import os
     import sys
     from tempfile import mktemp
-    from multiprocessing import Manager, Process
-
-    from ..utils.bids import write_bidsignore, write_derivative_description
-    from .parser import parse_args
+    from mriqc import config, messages
+    from mriqc.cli.parser import parse_args
 
     # Run parser
     parse_args()
 
+    _plugin = config.nipype.get_plugin()
+    if config.nipype.plugin in ("MultiProc", "LegacyMultiProc"):
+        from importlib import import_module
+        from multiprocessing import set_start_method
+        from contextlib import suppress
+
+        with suppress(RuntimeError):
+            set_start_method("forkserver")
+
+        Plugin = getattr(
+            import_module(f"nipype.pipeline.plugins.{config.nipype.plugin.lower()}"),
+            f"{config.nipype.plugin}Plugin",
+        )
+        _plugin = {
+            "plugin": Plugin(plugin_args=config.nipype.plugin_args),
+        }
```

Comment:
> I don't understand why you're modifying `_plugin` instead of just:

The idea is to actually initialize the Plugin (only in the case of some multiproc variant) before anything happens. The rationale is that the forkserver will be pinned up at this point instead of within the ``workflow.run()`` call, which happens later (after building up more memory).



> But supposing you need to initialize the plugin class, why not:

If I'm not wrong, that will load into vms the full plugins module. With the current approach I was hoping to only load the particular module and the base plugin module. Does that make sense?

Reply:
> > I don't understand why you're modifying `_plugin` instead of just:
> 
> The idea is to actually initialize the Plugin (only in the case of some multiproc variant) before anything happens. The rationale is that the forkserver will be pinned up at this point instead of within the `workflow.run()` call, which happens later (after building up more memory).

I think we may have a bad model of forkserver in our heads. I went to look up at what point the forkserver initialized its memory, whether at pool creation (which you seem to be assuming) or submission of the first job (what I thought). In fact, the forkserver loads up a new Python interpreter and pre-loads modules (defaulting only to `__main__`, so the variables that are available at the entrypoint). The objects in memory don't seem to matter, except they might be there for that first fork.

https://github.com/python/cpython/blob/main/Lib/multiprocessing/forkserver.py

If we want to be extra cautious, we could do the following instead:

```suggestion
    if config.nipype.plugin in ("MultiProc", "LegacyMultiProc"):
        import multiprocessing as mp
        import multiprocessing.forkserver

        with suppress(RuntimeError):
            mp.set_start_method("forkserver")
            mp.forkserver.ensure_running()
```

You do need to `set_start_method()` first, otherwise `ensure_running` will actually lock it in as `"fork"`. But this will definitely cause the `forkserver` to be initiated at this point while not requiring us to load in any nipype machinery.